contracts/vesting/src/contract.rs

```diff
@@ -0,0 +1,636 @@
+    //     env: Env,
+    //     sender: Address,
+    //     owner: Address,
+    //     amount: i128,
+    // ) -> Result<(), ContractError> {
+    //     sender.require_auth();
+
+    //     if amount <= 0 {
+    //         log!(&env, "Vesting: Burn from: Invalid burn amount");
+    //         panic_with_error!(env, ContractError::InvalidBurnAmount);
+    //     }
+
+    //     let allowance = get_allowances(&env, &(owner.clone(), sender.clone()));
+    //     if allowance < amount {
+    //         log!(&env, "Vesting: Burn from: Not enough allowance");
+    //         panic_with_error!(env, ContractError::NotEnoughBalance);
+    //     }
+
+    //     let new_allowance = allowance.checked_sub(amount).unwrap_or_else(|| {
+    //         log!(
+    //             &env,
+    //             "Vesting: Burn from: Critical error - allowance cannot be negative"
+    //         );
+    //         panic_with_error!(env, ContractError::Std);
+    //     });
+
+    //     let total_supply = get_vesting_total_supply(&env)
+    //         .checked_sub(amount)
+    //         .unwrap_or_else(|| {
+    //             log!(
+    //                 &env,
+    //                 "Vesting: Burn from: Critical error - total supply cannot be negative"
+    //             );
+    //             panic_with_error!(env, ContractError::Std);
+    //         });
+
+    //     update_vesting_total_supply(&env, total_supply);
+
+    //     let token_client = token_contract::Client::new(&env, &get_config(&env).token_info.address);
+    //     token_client.burn(&owner, &amount);
+
+    //     save_allowances(&env, &(owner, sender), new_allowance);
+
+    //     env.events()
+    //         .publish(("Burn from", "Burned tokens: "), amount);
+
+    //     Ok(())
+    // }
+
+    // fn send_to_contract_from(
+    //     env: Env,
+    //     sender: Address,
+    //     owner: Address,
+    //     contract: Address,
+    //     amount: i128,
+    // ) -> Result<(), ContractError> {
+    //     sender.require_auth();
+    //     if amount <= 0 {
+    //         log!(&env, "Vesting: Send to contract from: Invalid amount");
+    //         panic_with_error!(env, ContractError::InvalidTransferAmount);
+    //     }
+    //     //used to verify that the sender is authorized by the owner
+    //     let _ = get_allowances(&env, &(owner.clone(), sender.clone()));
+
+    //     let token_client = token_contract::Client::new(&env, &get_config(&env).token_info.address);
+    //     token_client.transfer(&owner, &contract, &amount);
+
+    //     env.events().publish(
+    //         (
+    //             "Send to contract from",
+    //             "Sent tokens to contract from account: from: {}, to: {}, amount: {}",
+    //         ),
+    //         (owner, contract, amount),
+    //     );
+
+    //     Ok(())
+    // }
+
+    fn update_minter(env: Env, sender: Address, new_minter: Address) {
+        if sender != get_minter(&env).address && sender != get_admin(&env) {
```

Comment:
Is this tested? 

```suggestion
        if sender != get_minter(&env).address || sender != get_admin(&env) {
```

Reply:
yes, both valid and invalid scenario are tested under `update_minter_works_correctly` and `update_minter_fails_when_not_authorized` respectively